indico/modules/rb/client/js/common/bookings/BookingDetails.jsx

```diff
@@ -448,8 +526,21 @@ class BookingDetails extends React.Component {
       booking: {id},
       actions: {changeBookingState},
     } = this.props;
+    const {acceptanceFormVisible} = this.state;
     this.setState({actionInProgress: action});
-    return changeBookingState(id, action, data).then(() => {
+    return changeBookingState(id, action, data).then(({error}) => {
+      if (error && error.message === 'prebooking_collision' && action === 'approve') {
+        if (acceptanceFormVisible) {
```

Comment:
I think it'd be safe to hide it unconditionally, since the setState will simply do nothing in that case. Nothing wrong with keeping the check though.